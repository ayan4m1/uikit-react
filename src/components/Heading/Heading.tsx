import * as React from 'react'
import classNames from 'classnames'
import { setClassNames } from '../../utils/set-class-names'

export class Heading extends React.Component<HeadingProps, any> {
  render() {
    const { id, style, children, line, element } = this.props

    const newChildren = line ? <span>{children}</span> : children

    return React.createElement(
      element ? element : 'h1',
      {
        id: id ? id : null,
        style: style ? style : null,
        className: this.setClassNames(),
      },
      newChildren ? newChildren : '',
    )
  }

  private setClassNames() {
    const { type, divider, bullet, line } = this.props

    return classNames({
      [`uk-heading-primary`]: type === 'primary',
      [`uk-heading-hero`]: type === 'primary',
      [`uk-heading-divider`]: !!divider,
      [`uk-heading-bullet`]: !!bullet,
      [`uk-heading-line`]: !!line,
      [`${setClassNames(this.props)}`]: true,
    })
  }
}
