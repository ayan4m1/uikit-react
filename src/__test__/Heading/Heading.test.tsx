import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Heading } from '../../components/Heading/Heading'

describe('<Heading>', () => {
  it('Should render without props', () => {
    const wrapper = renderer.create(<Heading>Test</Heading>)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('Should render with props', () => {
    const wrapper = renderer.create(
      <Heading id="test" style={{ marginTop: 10 }} element="h4">
        Test
      </Heading>,
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('Should render as primary', () => {
    const wrapper = renderer.create(<Heading type="primary">Test</Heading>)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('Should render as hero', () => {
    const wrapper = renderer.create(<Heading type="hero">Test</Heading>)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('Should render with bullet', () => {
    const wrapper = renderer.create(<Heading bullet={true}>Test</Heading>)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('Should render with divider', () => {
    const wrapper = renderer.create(<Heading divider={true}>Test</Heading>)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('Should render with line', () => {
    const wrapper = renderer.create(<Heading line={true}>Test</Heading>)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
