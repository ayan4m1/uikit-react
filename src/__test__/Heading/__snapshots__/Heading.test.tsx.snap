// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading> Should render as hero 1`] = `
<h1
  className="uk-heading-hero "
>
  Test
</h1>
`;

exports[`<Heading> Should render as primary 1`] = `
<h1
  className="uk-heading-primary "
>
  Test
</h1>
`;

exports[`<Heading> Should render with bullet 1`] = `
<h1
  className="uk-heading-bullet "
>
  Test
</h1>
`;

exports[`<Heading> Should render with divider 1`] = `
<h1
  className="uk-heading-divider "
>
  Test
</h1>
`;

exports[`<Heading> Should render with line 1`] = `
<h1
  className="uk-heading-line "
>
  <span>
    Test
  </span>
</h1>
`;

exports[`<Heading> Should render with props 1`] = `
<h4
  className=""
  id="test"
  style={
    Object {
      "marginTop": 10,
    }
  }
>
  Test
</h4>
`;

exports[`<Heading> Should render without props 1`] = `
<h1
  className=""
>
  Test
</h1>
`;
